infrastructure/crypto/src/ristretto/pedersen.rs

```diff
@@ -57,12 +58,19 @@ impl Default for PedersenBaseOnRistretto255 {
     }
 }
 
+impl Default for &PedersenBaseOnRistretto255 {
+    fn default() -> Self {
+        &DEFAULT_RISTRETTO_PEDERSON_BASE
+    }
+}
+
 lazy_static! {
     pub static ref DEFAULT_RISTRETTO_PEDERSON_BASE: PedersenBaseOnRistretto255 = PedersenBaseOnRistretto255::default();
 }
 
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
 pub struct PedersenOnRistretto255 {
+    #[serde(skip)]
```

Comment:
This will be broken if we ever use a non-"default" Pedersen Base and is just inviting weird bugs in future.

We currently avoid this issue by using `as_public_key` and `from_public_key` in the `HomomorphicCommitmentFactory` struct, which always ensure that the correct base is used.

Can you figure out a way to use the factory to de-serialise Commitments (and store the public keys). Essentially, Commitments themselves cannot be serialised.